local-cluster/tests/local_cluster.rs

```diff
@@ -1137,14 +1137,27 @@ fn test_fork_choice_refresh_old_votes() {
 
             info!("Opened blockstores");
 
-            // Get latest votes
+            // Get latest votes. We additionally check to make sure the vote has landed in
+            // blockstore. This is important because if we were the leader for the block there
+            // is a possibility of voting before broadcast has inserted in blockstore.
             let (lighter_fork_latest_vote, _) = last_vote_in_tower(
                 &lighter_fork_ledger_path,
                 &context.lighter_fork_validator_key,
             )
             .unwrap();
             let (heaviest_fork_latest_vote, _) =
                 last_vote_in_tower(&heaviest_ledger_path, &context.heaviest_validator_key).unwrap();
+            while lighter_fork_blockstore
+                .meta(lighter_fork_latest_vote)
+                .unwrap()
+                .is_none()
+                || heaviest_blockstore
+                    .meta(heaviest_fork_latest_vote)
+                    .unwrap()
+                    .is_none()
```

Comment:
I think we should also wait for the block to be full, might be easier to just call `while !blockstore.is_full()`